testing/sdk_auth.py

```diff
@@ -208,33 +210,78 @@ def install(self) -> Dict[str, Any]:
             log.info("Found installed KDC app, destroying it first")
             sdk_marathon.destroy_app(self.app_definition["id"])
 
+        # (re-)create a service account for the KDC service
+        sdk_security.create_service_account("kdc-admin", "kdc-admin")  # Account name  # Secret name
+        sdk_security._grant(
+            "kdc-admin",
+            "dcos:secrets:default:%252F*",
+            "Create any secret in the root path",
+            "create",
+        )
+        sdk_security._grant(
+            "kdc-admin",
+            "dcos:secrets:default:%252F*",
+            "Update any secret in the root path",
+            "update",
+        )
+
         log.info("Installing KDC Marathon app")
         sdk_marathon.install_app(self.app_definition)
         log.info("KDC app installed successfully")
 
+        log.info("Waiting for KDC web API endpoint to become available")
+        self.__wait_for_kdc_api()
+        log.info("KDC web API is now available")
+
         return _get_kdc_task(self.app_definition["id"])
 
-    def __run_kadmin(self, options: List[str], cmd: str, args: List[str]) -> None:
+    def __wait_for_kdc_api(self):
+        """
+        Keeps polling the KDC Web endpoint until it becomes available
         """
-        Invokes Kerberos' kadmin binary inside the container to run some command.
-        :param options: A list of options given to kadmin.
-        :param cmd: The name of the sub command to run.
-        :param args: A list of arguments passed to the sub command. This should also include any flags
-                     needed to be set for the sub command.
+
+        @retrying.retry(stop_max_attempt_number=30, wait_fixed=2000)
+        def probe():
+            sdk_cmd.cluster_request(
+                "GET", "{}/".format(self.get_service_path()), raise_on_error=True
+            )
+
+        return probe()
+
+    def __kdc_api(self, method: str, action: str, json: dict) -> dict:
+        """
+        Invokes a KDC API command to the remote endpoint
+        :param method: 'get' or 'post'
+        :param url: The API action to perform
+        :param request: The JSON payload to send
+        :return (dict):
         :raises a generic Exception if the invocation fails.
         """
-        kadmin_cmd = "/usr/sbin/kadmin {options} {cmd} {args}".format(
-            options=" ".join(options), cmd=cmd, args=" ".join(args)
+        url = "{}/api/{}".format(self.get_service_path(), action)
+        log.info("Performing KDC API {method} query to: {url}".format(method=method, url=url))
+
+        # Place the web request to cluster
```

Comment:
Please remove this comment, it is stating the obvious, given the preceding log statement and the self-explanatory call below.